Bioinf_Utility/bioinf_utility.py

```diff
@@ -0,0 +1,173 @@
+        if operation in operations:
+            return operations[operation](sequence)
+
+    elif len(seq_oper) > 2:
+        operation, sequences = seq_oper[-1], seq_oper[:-1]
+        results = []
+        for sequence in sequences:
+            if operation in operations:
+                operation_result = operations[operation](sequence)
+                results.append(operation_result)
+            else:
+                return "Wrong operation"
+        return results
+
+    else:
+        return "Wrong amount of arguments"
+
+
+def filter_fastq(
+    input_fastq,
+    output_fastq=None,
+    gc_bounds=(0, 100),
+    length_bounds=(0, 2**32),
+    quality_threshold=0,
+):
+    """
+    Filters FASTQ sequences based on GC content, sequence length, and quality
+    score.
+
+    This function reads sequences from an input FASTQ file and writes the
+    filtered sequences to an output FASTQ file in a 'filtered' directory.
+    Sequences are filtered according to specified GC content bounds, sequence
+    length bounds, and a minimum quality threshold.
+
+    Parameters:
+    input_fastq (str):
+        Path to the input FASTQ file containing the sequences to be filtered.
+
+    output_fastq (str, optional):
+        Name of the output file to which filtered sequences will be saved.
+        If not specified, the output file will be saved with the same base name
+        as the input file.
+        File will be saved to the 'filtered' subfolder in FASTQ format.
+
+    gc_bounds (tuple or int, optional):
+        A tuple specifying the minimum and maximum GC-content percentages
+        (default is (0, 100)). If an integer is provided, it is treated as the
+        maximum GC content with a minimum of 0.
+
+    length_bounds (tuple or int, optional):
+        A tuple specifying the minimum and maximum sequence lengths
+        (default is (0, 2**32)). If an integer is provided, it is treated as
+        the maximum sequence length with a minimum of 0.
+
+    quality_threshold (int, optional):
+        The minimum acceptable quality score for sequences (default is 0).
+        All sequences must have a quality score greater than or equal to this
+        threshold to pass the filter.
+
+    Returns:
+        None: Writes the filtered sequences to the specified output FASTQ file.
+        If any of the input parameters are invalid, returns the error message.
+
+    Example:
+        To filter sequences from a FASTQ file based on specific criteria:
+
+        filter_fastq(
+            "data/example_fastq.fastq",
+            "each_filter_output.fastq",
+            (40, 80),
+            (10, 20),
+            35
+        )
+
+        # This will write the filtered sequences to
+        # 'filtered/each_filter_output.fastq'.
+    """
+    if type(gc_bounds) is int and gc_bounds > 0:
+        gc_bounds = (0, gc_bounds)
+    elif type(gc_bounds) is not tuple and len(gc_bounds) != 2:
+        return "Wrong value for the gc_bounds argument"
```

Comment:
вместо проверки через type() можно использовать isinstance(). Например: 

```suggestion
    elif not isinstance(gc_bounds, tuple) and len(gc_bounds) != 2:
        return "Wrong value for the gc_bounds argument"
```